$MirOS: ports/devel/subversion/patches/patch-subversion_libsvn_repos_repos_c,v 1.1 2007/01/09 22:37:00 tg Exp $

	Help sucky operating systems to be less sucky.

	In general, if the command
	$ env -i LC_CTYPE=en_US.UTF-8 /usr/bin/locale >&-
	fails - you can check this for example with
	$ env -i LC_CTYPE=nonexistant /usr/bin/locale >&-
	then you need to take some action, e.g.
	# localedef -c -i en_US -f UTF-8 en_US.UTF-8
	(or with sudo), if that works.

--- subversion/libsvn_repos/repos.c.orig	2006-08-23 15:37:25.000000000 +0200
+++ subversion/libsvn_repos/repos.c	2007-09-15 01:22:05.000000000 +0200
@@ -1038,11 +1038,23 @@ create_hooks(svn_repos_t *repos, apr_poo
       HOOKS_ENVIRONMENT_TEXT
       "# "
       APR_EOL_STR
+      "# Its standard output and standard error are not displayed"
+      APR_EOL_STR
+      "# to the client either."
+      APR_EOL_STR
+      "# "
+      APR_EOL_STR
       "# Here is an example hook script, for a Unix /bin/sh interpreter."
       APR_EOL_STR
       PREWRITTEN_HOOKS_TEXT
       APR_EOL_STR
       APR_EOL_STR
+      "# If this breaks, your operating system is fucked up anyway."
+      APR_EOL_STR
+      "# Consider the documentation on how to create this locale, then."
+      APR_EOL_STR
+      "export LC_CTYPE=en_US.UTF-8"
+      APR_EOL_STR
       "REPOS=\"$1\""
       APR_EOL_STR
       "REV=\"$2\""
